### PR TITLE
Add presigned URL download support to S3 Transfer Manager

### PR DESCRIPTION
**Pull request by jencymaryjoseph**
## Motivation and Context
Adds presigned URL download support to S3 Transfer Manager with `downloadFileWithPresignedUrl` and `downloadWithPresignedUrl` methods. This enables customers to download objects using pre-signed URLs through the Transfer Manager with multipart download support and progress tracking.

## Modifications

- `S3TransferManager`: Added `downloadFileWithPresignedUrl` (with consumer builder variant) and `downloadWithPresignedUrl` public API methods with javadoc and code snippets.
- `GenericS3TransferManager`: Implemented presigned URL download logic with multipart support and progress tracking using wrapForNonSerialFileDownload.
- `S3IntegrationTestBase`: Added `tmNonMultipartJava` (non-multipart transfer manager) and `presignedUrlTransferManagers`() method source for presigned URL integration tests. CRT client is commented out with a TODO to enable when CRT presigned URL support is available.

## Testing

- `S3TransferManagerPresignedUrlDownloadTest`: Unit tests covering happy path (toFile/toBytes), consumer builder, cancellation forwarding, error propagation, and null validation.
- `S3TransferManagerPresignedUrlDownloadIntegrationTest`: Integration tests parameterized by transfer manager type (nonMultipart/multipart) and file size (5MB/16MB), verifying data integrity with checksums for both downloadFileWithPresignedUrl and downloadWithPresignedUrl.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/aws/aws-sdk-java-v2/blob/master/CONTRIBUTING.md) document
- [x] Local run of `mvn install` succeeds
- [x] My code follows the code style of this project
- [ ] My change requires a change to the Javadoc documentation
- [ ] I have updated the Javadoc documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] I have added a changelog entry. Adding a new entry must be accomplished by running the `scripts/new-change` script and following the instructions. Commit the new file created by the script in `.changes/next-release` with your changes.
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: services-custom/s3-transfer-manager/src/it/java/software/amazon/awssdk/transfer/s3/S3TransferManagerPresignedUrlDownloadIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+import software.amazon.awssdk.transfer.s3.model.PresignedDownloadRequest;
+import software.amazon.awssdk.transfer.s3.progress.LoggingTransferListener;
+import software.amazon.awssdk.utils.Md5Utils;
+
+public class S3TransferManagerPresignedUrlDownloadIntegrationTest extends S3IntegrationTestBase {
+    private static final String BUCKET = temporaryBucketName(S3TransferManagerPresignedUrlDownloadIntegrationTest.class);
+    private static final String SMALL_KEY = "small-key";
+    private static final String LARGE_KEY = "large-key";
+    private static final int SMALL_OBJ_SIZE = 5 * 1024 * 1024;
+    private static final int LARGE_OBJ_SIZE = 16 * 1024 * 1024;
```
Comment (RanVaknin) [inline services-custom/s3-transfer-manager/src/it/java/software/amazon/awssdk/transfer/s3/S3TransferManagerPresignedUrlDownloadIntegrationTest.java:55]: Can you please run one off tests (you dont have to commit those to the PR) against larger files like 1GB and 10GB. Trying to see if size would reveal any edge cases.

File: services-custom/s3-transfer-manager/src/test/java/software/amazon/awssdk/transfer/s3/internal/S3TransferManagerPresignedUrlDownloadTest.java [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+                                         mock(UploadDirectoryHelper.class),
+                                         mock(TransferManagerConfiguration.class),
+                                         mock(DownloadDirectoryHelper.class));
+    }
+
+    @AfterEach
+    public void methodTeardown() {
+        tm.close();
+    }
+
```
Comment (RanVaknin) [inline services-custom/s3-transfer-manager/src/test/java/software/amazon/awssdk/transfer/s3/internal/S3TransferManagerPresignedUrlDownloadTest.java:74]: We are missing tests that verify that the progress tracking works for presigned URL downloads. 
See `S3TransferManagerListenerTest.java` as an example. (you need to make sure the callback hooks are firing for the provided transfer listener)

File: services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/internal/GenericS3TransferManager.java [hunk 3]
```diff
@@ -597,6 +599,76 @@ public final Copy copy(CopyRequest copyRequest) {
...
+            CompletableFutureUtils.forwardExceptionTo(returnFuture, future);
+            CompletableFutureUtils.forwardTransformedResultTo(future, returnFuture,
+                                                              res -> CompletedFileDownload.builder()
+                                                                                          .response(res)
+                                                                                          .build());
+        } catch (Throwable throwable) {
+            returnFuture.completeExceptionally(throwable);
+        }
+
+        return new DefaultFileDownload(returnFuture, progressUpdater.progress(), () -> null, null);
```
Comment (RanVaknin) [inline services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/internal/GenericS3TransferManager.java]: The callback shouldn't just return null. We should throw an exception instead if customers call .pause();

```java
  () -> { throw new UnsupportedOperationException("pause is not supported...."); }
```

File: services-custom/s3-transfer-manager/src/it/java/software/amazon/awssdk/transfer/s3/S3IntegrationTestBase.java [hunk 4]
```diff
@@ -54,12 +54,14 @@ public class S3IntegrationTestBase extends AwsTestBase {
      */
     protected static S3Client s3;
 
-    protected static S3AsyncClient s3Async;
+    protected static S3AsyncClient s3Async;;
```
Comment (RanVaknin) [inline services-custom/s3-transfer-manager/src/it/java/software/amazon/awssdk/transfer/s3/S3IntegrationTestBase.java]: fix typo

File: services-custom/s3-transfer-manager/src/it/java/software/amazon/awssdk/transfer/s3/S3IntegrationTestBase.java [hunk 5]
```diff
@@ -182,4 +195,11 @@ static Stream<Arguments> transferManagers() {
             Arguments.of(tmJava));
     }
 
+    static Stream<Arguments> presignedUrlTransferManagers() {
```
Comment (RanVaknin) [inline services-custom/s3-transfer-manager/src/it/java/software/amazon/awssdk/transfer/s3/S3IntegrationTestBase.java:198]: Is this used somewhere?

**Comment by RanVaknin** [comment]
We need to add the new methods to `DelegatingS3TransferManager.java` so when someone subclasses that they would get forwarded to the delegate and not hit the default `unsupportedOperationException`

File: services-custom/s3-transfer-manager/src/it/java/software/amazon/awssdk/transfer/s3/S3TransferManagerPresignedUrlDownloadIntegrationTest.java (hunk 1 above)
Comment (jencymaryjoseph) [reply services-custom/s3-transfer-manager/src/it/java/software/amazon/awssdk/transfer/s3/S3TransferManagerPresignedUrlDownloadIntegrationTest.java:55]: Tested for 1GB download, looks good. Will do more tests during performance tests.

**Review by RanVaknin — CHANGES_REQUESTED** [review]

File: services-custom/s3-transfer-manager/src/it/java/software/amazon/awssdk/transfer/s3/S3IntegrationTestBase.java (hunk 5 above)
Comment (jencymaryjoseph) [reply services-custom/s3-transfer-manager/src/it/java/software/amazon/awssdk/transfer/s3/S3IntegrationTestBase.java:198]: Missed the @MethodSource reference, added now. It's used by the parameterized tests in `S3TransferManagerPresignedUrlDownloadIntegrationTest`.

File: services-custom/s3-transfer-manager/src/test/java/software/amazon/awssdk/transfer/s3/internal/S3TransferManagerPresignedUrlDownloadTest.java (hunk 2 above)
Comment (jencymaryjoseph) [reply services-custom/s3-transfer-manager/src/test/java/software/amazon/awssdk/transfer/s3/internal/S3TransferManagerPresignedUrlDownloadTest.java:74]: Will add these tests in an upcoming PR, working on it.

**Review by RanVaknin — APPROVED** [review]
